feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/README.md

```diff
@@ -4,33 +4,36 @@
 
 BGP Graceful Restart
 
-## Procedure
+## Topology
+Follwing connections:
+ *   ATE port-1 <--> DUT port-1 
+ *   ATE port-2 <--> DUT port-2
 
-*   Establish eBGP sessions between:
-    *   ATE port-1 and DUT port-1
-    *   ATE port-2 and DUT port-2
-    *   Configure allow route-policy under BGP peer-group address-family
-*   Validate received capabilities at DUT and ATE reflect support for graceful
-    restart.
-*   For IPv4 and IPv6 routes:
-    *   (Restarting speaker) Advertise prefixes between the ATE ports, through
-        the DUT. Trigger DUT session restart by disconnecting TCP session
-        between DUT and ATE (this may be achieved by using an ACL), determine
-        that packets are:
-        *   Forwarded between ATE port-1 and DUT port-1 for the duration of the
-            specified stale routes time.
-        *   Dropped after the stale routes timer has expired.
-        *   Forwarded again between ATE port-1 and DUT port-1 after the session
-            is re-established.
-    *   (Receiving speaker) Advertise prefixes between the ATE ports, through
-        the DUT. Trigger session restart by disconnecting the BGP session from
-        ATE port-2.
-        *   Ensure that prefixes are propagated to ATE port-2 during the
-            restart.
-        *   Ensure that traffic can be forwarded between ATE port-1 and ATE
-            port-2 during stale routes time.
-        *   Ensure that prefixes are withdrawn, and traffic cannot be forwarded
-            between ATE port-1 and port-2 after the stale routes time expires.
+## Procedure
+ *   Configure EBGP peering between ATE:Port1 and DUT:Port1
+ *   Configure IBGP peering between ATE:Port2 and DUT:Port2
+ *   Ensure that the EBGP and IBGP peering are setup for IPv4-Unicast and IPv6-unicast AFI-SAFIs. Total 2xpeer-groups (1 per protocol) with 1 BGP session each.  
+ *   Enable `Graceful-Restart` capability at the `Peer-Group` level.
+ *   Ensure that the `restart-time` and the `stale-routes-time` are configured at the `Global` level
+ *   Configure allow route-policy under BGP peer-group address-family
+ *   Validate received capabilities at DUT and ATE reflect support for graceful
+     restart.
+ *   TestCase - Restarting DUT speaker 
+   * Advertise prefixes between the ATE ports, through the DUT. 
+   * Trigger DUT session restart by killing the BGP process in the DUT. Please use the `gNOI.killProcessRequest_Signal_Term` as per [gNOI_proto](https://github.com/openconfig/gnoi/blob/main/system/system.proto#L326).
```

Comment:
Recomend using gNOI for session clear: https://github.com/openconfig/gnoi/blob/main/bgp/bgp.proto#L35

Reply:
But the goal here is to test BGP GR in the DUT in restarting mode. This requires the BGP process in the DUT to gracefully go down. Clear BGP Neighbor wouldn't do that but [Signal_Term](https://github.com/openconfig/gnoi/blob/main/system/system.proto#L326) would. Please check the conversation in b/275111471

Reply:
Ok, I think I get it now. The kill process `term` method is expected to gracefully restart the BGP process on the DUT.  The DUT is expected to stop it's  BGP TCP session with the ATE, preserve FIB state for BGP learned prefixes and start a new BGP TCP following https://www.rfc-editor.org/rfc/rfc4724.html#section-4.1.  

Seems like we need to ensure traffic is flowing using  prefixes learned from the ATE  during the restart to confirm FIB entries are maintained on the DUT and there is no packet loss during the graceful restart.

Can you add all this text into the test plan?

Reply:
Updated summary section. PTAL